auth/maintenance_socket_role_manager.cc

```diff
@@ -109,4 +110,12 @@ future<> maintenance_socket_role_manager::remove_attribute(std::string_view role
     return operation_not_supported_exception("REMOVE ATTRIBUTE");
 }
 
+future<std::vector<cql3::description>> maintenance_socket_role_manager::describe_role_grants() {
+    return operation_not_supported_exception<std::vector<cql3::description>>("DESCRIBE SCHEMA WITH INTERNALS");
```

Comment:
Oh, there is an implicit conversion from an exception to a failed future?? I'm not sure I am a fan - please wrap it with `make_ready_future` to be more explicit.

Reply:
The conversion is explicit. `operation_not_supported_exception` is an auxiliary function used in the file to generate consistent exception messages:

```cpp
template<typename T = void>
future<T> operation_not_supported_exception(std::string_view operation) {
    return make_exception_future<T>(
        std::runtime_error(fmt::format("role manager: {} operation not supported through maintenance socket", operation)));
}
```

Reply:
Oof, that's really confusing because it looks like a constructor. However, it's a preexisting problem and not a huge one - so it's fine to keep this code as it is.